### PR TITLE
Adding initial pack files

### PR DESCRIPTION
**Pull request by kedark3**

### CODE REVIEW

**Comment by kedark3** [comment]
@john-dupuy Please review.

**Comment by kedark3** [comment]
Also, do I need to add package.json?

File: etc/resource.d/osp.cfg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+$OSP_AUTH_URL$=http://osp.example.com:5000/v3
+$OSP_TENANT$=admin
+$OSP_USER$=admin
+$OSP_PASSWORD$=password
```
Comment (john-dupuy) [inline etc/resource.d/osp.cfg]: new line needed here

File: pack/commands.cfg [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#      OpenStack standard check
+#      Need  https://github.com/kedark3/check-osp
+#
+# -----------------------------------------------------------------
+
+
+## Check commands for openstack
+define command{
+       command_name     check_vm_count
+       command_line     $PLUGINSDIR$/check_osp_main.py -R $HOSTALIAS$ -u $_HOSTOSP_USER$ -p $_HOSTOSP_PASSWORD$ -t $_HOSTOSP_TENANT$ -m vm_count -c $_HOSTRHV_VM_COUNT_CRIT$ -w $_HOSTRHV_VM_COUNT_WARN$
```
Comment (john-dupuy) [inline pack/commands.cfg]: You're using some `RHV` macros for the warning and critical values.

File: pack/commands.cfg [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+}
+define command{
+       command_name     check_locked_disks_count
+       command_line     $PLUGINSDIR$/check_osp_main.py -R $HOSTALIAS$ -u $_HOSTOSP_USER$ -p $_HOSTOSP_PASSWORD$ -t $_HOSTOSP_TENANT$ -m keypair_count -c $_HOSTRHV_LOCKED_DISKS_CRIT$ -w $_HOSTRHV_LOCKED_DISKS_WARN$
```
Comment (john-dupuy) [inline pack/commands.cfg]: Same issue here.

File: pack/commands.cfg [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+}
+define command{
+       command_name     check_locked_disks_count
+       command_line     $PLUGINSDIR$/check_osp_main.py -R $HOSTALIAS$ -u $_HOSTOSP_USER$ -p $_HOSTOSP_PASSWORD$ -t $_HOSTOSP_TENANT$ -m volume_count -c $_HOSTRHV_LOCKED_DISKS_CRIT$ -w $_HOSTRHV_LOCKED_DISKS_WARN$
```
Comment (john-dupuy) [inline pack/commands.cfg]: And here

File: pack/osp-wrapanapi.pack [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+"description": "Checks for Red Hat OpenStack using wrapanapi",
+
+"path":"cloud/",
+
+"templates": ["OSPmanager"],
+
+"macros":{
+         "_OSP_VM_COUNT_WARN": {"type":"integer",
+                             "description":"Number of VMs in OSP"
```
Comment (john-dupuy) [inline pack/osp-wrapanapi.pack]: I would put `Warning number of VMs in OSP`

File: pack/osp-wrapanapi.pack [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+                             },
+         "_OSP_VM_COUNT_CRIT": {"type":"integer",
+                             "description":"Number of VMs in OSP"
```
Comment (john-dupuy) [inline pack/osp-wrapanapi.pack]: Same thing here and below `Critical number of VMs in OSP`

File: pack/services/osp_services.cfg [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+   register       0
+   host_name	  osp
+   check_command  check_volume_count
+}
+define service{
+   service_description          Hosts-Services-Status
+   use            osp-service
+   register       0
+   host_name     osp
+   check_command  check_services_running
```
Comment (john-dupuy) [inline pack/services/osp_services.cfg]: This is call `check_host_services_running` in the command you've defined above. I would call it something unique to distinguish it from the rhv-check.

File: pack/commands.cfg [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+define command{
+       command_name     check_vm_count
+       command_line     $PLUGINSDIR$/check_osp_main.py -R $HOSTALIAS$ -u $_HOSTOSP_USER$ -p $_HOSTOSP_PASSWORD$ -t $_HOSTOSP_TENANT$ -m vm_count -c $_HOSTRHV_VM_COUNT_CRIT$ -w $_HOSTRHV_VM_COUNT_WARN$
+}
+define command{
+       command_name     check_locked_disks_count
+       command_line     $PLUGINSDIR$/check_osp_main.py -R $HOSTALIAS$ -u $_HOSTOSP_USER$ -p $_HOSTOSP_PASSWORD$ -t $_HOSTOSP_TENANT$ -m keypair_count -c $_HOSTRHV_LOCKED_DISKS_CRIT$ -w $_HOSTRHV_LOCKED_DISKS_WARN$
+}
+define command{
+       command_name     check_locked_disks_count
```
Comment (john-dupuy) [inline pack/commands.cfg]: There are two commands called `check_locked_disks_count`, I think this one should be `check_osp_volumes_count`

File: pack/commands.cfg [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+#
+#      OpenStack standard check
+#      Need  https://github.com/kedark3/check-osp
+#
+# -----------------------------------------------------------------
+
+
+## Check commands for openstack
+define command{
+       command_name     check_vm_count
```
Comment (john-dupuy) [inline pack/commands.cfg]: For these commands I would distinguish them from the RHV checks, so we know they are specific to OSP

File: pack/commands.cfg [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+       command_line     $PLUGINSDIR$/check_osp_main.py -R $HOSTALIAS$ -u $_HOSTOSP_USER$ -p $_HOSTOSP_PASSWORD$ -t $_HOSTOSP_TENANT$ -m vm_count -c $_HOSTRHV_VM_COUNT_CRIT$ -w $_HOSTRHV_VM_COUNT_WARN$
+}
+define command{
+       command_name     check_locked_disks_count
```
Comment (john-dupuy) [inline pack/commands.cfg]: Should be called `check_osp_keypair_count`?

File: pack/templates.cfg [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+   register     0
+}
+
+# OSP system template
+define host{
+    name        OSP
+    use         generic-host
+    register    0
+
+    _OSP_USER    $$OSP_USER$$
```
Comment (john-dupuy) [inline pack/templates.cfg]: Why two `$$` around this macro?

File: pack/commands.cfg [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+
+
+## Check commands for openstack
+define command{
+       command_name     check_osp_vm_count
+       command_line     $PLUGINSDIR$/check_osp_main.py -R $HOSTALIAS$ -u $_HOSTOSP_USER$ -p $_HOSTOSP_PASSWORD$ -t $_HOSTOSP_TENANT$ -m vm_count -c $_HOSTOSP_VM_COUNT_CRIT$ -w $_HOSTOSP_VM_COUNT_WARN$
+}
+define command{
+       command_name     check_osp_keypair_count
+       command_line     $PLUGINSDIR$/check_osp_main.py -R $HOSTALIAS$ -u $_HOSTOSP_USER$ -p $_HOSTOSP_PASSWORD$ -t $_HOSTOSP_TENANT$ -m keypair_count -c $_HOSTOSP_LOCKED_DISKS_CRIT$ -w $_HOSTOSP_LOCKED_DISKS_WARN$
```
Comment (john-dupuy) [inline pack/commands.cfg]: This is still using the `LOCKEDDISK` macro

File: pack/commands.cfg [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+}
+define command{
+       command_name     check_osp_volumes_count
+       command_line     $PLUGINSDIR$/check_osp_main.py -R $HOSTALIAS$ -u $_HOSTOSP_USER$ -p $_HOSTOSP_PASSWORD$ -t $_HOSTOSP_TENANT$ -m volume_count -c $_HOSTOSP_LOCKED_DISKS_CRIT$ -w $_HOSTOSP_LOCKED_DISKS_WARN$
```
Comment (john-dupuy) [inline pack/commands.cfg]: Same issue here

File: pack/commands.cfg [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+define command{
+       command_name     check_osp_vm_count
+       command_line     $PLUGINSDIR$/check_osp_main.py -R $HOSTALIAS$ -u $_HOSTOSP_USER$ -p $_HOSTOSP_PASSWORD$ -t $_HOSTOSP_TENANT$ -m vm_count -c $_HOSTOSP_VM_COUNT_CRIT$ -w $_HOSTOSP_VM_COUNT_WARN$
+}
+define command{
+       command_name     check_osp_keypair_count
+       command_line     $PLUGINSDIR$/check_osp_main.py -R $HOSTALIAS$ -u $_HOSTOSP_USER$ -p $_HOSTOSP_PASSWORD$ -t $_HOSTOSP_TENANT$ -m keypair_count -c $_HOSTOSP_LOCKED_DISKS_CRIT$ -w $_HOSTOSP_LOCKED_DISKS_WARN$
+}
+define command{
+       command_name     check_osp_volumes_count
```
Comment (john-dupuy) [inline pack/commands.cfg]: To maintain consistency with the other count checks, I think `volume` should be singular, i.e. `check_osp_volume_count`

File: pack/services/osp_services.cfg [hunk 15]
```diff
@@ -0,0 +1,28 @@
+define service{
+   service_description          VMs-Count
```
Comment (john-dupuy) [inline pack/services/osp_services.cfg]: I would call this `VM-Count` to match the check_command name

File: pack/commands.cfg [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+       command_name     check_osp_vm_count
+       command_line     $PLUGINSDIR$/check_osp_main.py -R $HOSTALIAS$ -u $_HOSTOSP_USER$ -p $_HOSTOSP_PASSWORD$ -t $_HOSTOSP_TENANT$ -m vm_count -c $_HOSTOSP_VM_COUNT_CRIT$ -w $_HOSTOSP_VM_COUNT_WARN$
+}
+define command{
+       command_name     check_osp_keypair_count
+       command_line     $PLUGINSDIR$/check_osp_main.py -R $HOSTALIAS$ -u $_HOSTOSP_USER$ -p $_HOSTOSP_PASSWORD$ -t $_HOSTOSP_TENANT$ -m keypair_count -c $_HOSTOSP_KEYPAIR_COUNT_CRIT$ -w $_HOSTOSP_KEYPAIR_COUNT_WARN$
+}
+define command{
+       command_name     check_osp_volume_count
+       command_line     $PLUGINSDIR$/check_osp_main.py -R $HOSTALIAS$ -u $_HOSTOSP_USER$ -p $_HOSTOSP_PASSWORD$ -t $_HOSTOSP_TENANT$ -m volume_count -c $_HOSTOSP_VM_COUNT_CRIT$ -w $_HOSTOSP_VM_COUNT_WARN$
```
Comment (john-dupuy) [inline pack/commands.cfg]: This is using `VOLUME_COUNT`, is that intentional?

**Pull request merged**